test/jobs/job-status-pagination.ts

```diff
@@ -0,0 +1,209 @@
+
+  before(async function () {
+    // use a DEFAULT_PAGE size of 10 for these tests
+    ({ defaultPageSize } = env);
+    env.defaultPageSize = 10;
+
+    // Generate some links for the job - need to save the job after each link is added to
+    // circumvent ordering problems with sqlite
+    const jTrx = await db.transaction();
+    await aJob.save(jTrx);
+    jTrx.commit();
+    for (let i = 1; i < 51; i++) {
+      links.push(
+        new JobLink({
+          href: `http://example.com/${i}`,
+          title: `Example ${i}`,
+          type: i % 2 === 0 ? 'text/plain' : 'text/ornate',
+          rel: 'data',
+        }),
+      );
+    }
+
+    for (const link of links) {
+      const trx = await db.transaction();
+      aJob.addLink(
+        link,
+      );
+      await aJob.save(trx);
+      trx.commit();
+    }
+  });
+
+  after(function () {
+    env.defaultPageSize = defaultPageSize;
+  });
+
+  const jobID = aJob.requestId;
+  describe('when `page` parameter is set', function () {
+    describe('and the page is a valid page', function () {
+      hookJobStatus({ jobID, username: 'joe', query: { page: 2 } });
+      it('shows the corresponding page of results', function () {
+        const job = new Job(JSON.parse(this.res.text));
+        const outputLinks = job.getRelatedLinks('data');
+        expect(areJobLinksEqual(links.slice(10, 20), outputLinks)).to.equal(true);
+      });
+    });
+
+    describe('and the page is a not a valid page', function () {
+      hookJobStatus({ jobID, username: 'joe', query: { page: 0 } });
+      it('returns a 400 HTTP Bad request response', function () {
+        expect(this.res.statusCode).to.equal(400);
+      });
+
+      it('returns a JSON error response', function () {
+        const response = JSON.parse(this.res.text);
+        expect(response).to.eql({
+          code: 'harmony.RequestValidationError',
+          description: 'Error: Parameter "page" is invalid. Must be an integer greater than or equal to 1.',
+        });
+      });
+    });
+
+    describe('and the page is not an integer', function () {
+      hookJobStatus({ jobID, username: 'joe', query: { page: 1.5 } });
+      it('returns a 400 HTTP Bad request response', function () {
+        expect(this.res.statusCode).to.equal(400);
+      });
+
+      it('returns a JSON error response', function () {
+        const response = JSON.parse(this.res.text);
+        expect(response).to.eql({
+          code: 'harmony.RequestValidationError',
+          description: 'Error: Parameter "page" is invalid. Must be an integer greater than or equal to 1.',
+        });
+      });
+    });
+  });
+
+  describe('when `page` parameter and `limit` are not set', function () {
+    hookJobStatus({ jobID, username: 'joe', query: {} });
+    it('shows the first page of results with the default page size', function () {
+      const job = new Job(JSON.parse(this.res.text));
+      const outputLinks = job.getRelatedLinks('data');
```

Comment:
Note that this will need to change for Vinny's 483 PR if he's going to change getRelatedLinks to make a database call.

Reply:
He is no longer making that change.